specification/marketplacecatalog/resource-manager/Solutions/preview/2023-03-01-preview/solutions.json

```diff
@@ -0,0 +1,482 @@
+            "description": "*** Error Responses: ***\n\n * 400 - In case request malformed.\n\n * 500 - In case internal error.",
+            "x-ms-error-response": true,
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/mysolutions/{id}": {
+      "get": {
+        "tags": [
+          "UserSolutions"
+        ],
+        "description": "Returns a user solution",
+        "operationId": "GetUserSolution",
+        "x-ms-examples": {
+          "GetUserSolution": {
+            "$ref": "./examples/GetUserSolution.json"
+          }
+        },
+        "parameters": [
+          {
+            "type": "string",
+            "name": "id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "type": "string",
+            "name": "api-version",
+            "in": "query",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Solution"
+            }
+          },
+          "default": {
+            "x-nullable": false,
+            "description": "*** Error Responses: ***\n\n * 400 - In case request malformed.\n\n * 500 - In case internal error.",
+            "x-ms-error-response": true,
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "UserSolutions"
+        ],
+        "description": "Removes a user solution",
+        "operationId": "RemoveUserSolution",
+        "x-ms-examples": {
+          "RemoveUserSolution": {
+            "$ref": "./examples/RemoveUserSolution.json"
+          }
+        },
+        "parameters": [
+          {
+            "type": "string",
+            "name": "id",
+            "in": "path",
+            "required": true
+          },
+          {
+            "type": "string",
+            "name": "api-version",
+            "in": "query",
+            "required": true
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "x-nullable": false,
+            "description": "*** Error Responses: ***\n\n * 400 - In case request malformed.\n\n * 500 - In case internal error.",
+            "x-ms-error-response": true,
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/mysolutions/{id}/rename": {
```

Comment:
PUT must be used to update the resource.

Reply:
Can we use PATCH?

Reply:
We only want to update the name

Reply:
Thanks for sharing.